util/chplenv/printchplenv.py

```diff
@@ -258,20 +263,22 @@ def compute_internal_values():
     if chpl_re2.get() != 'none':
         link_args_3p.extend(chpl_re2.get_link_args())
 
-    # Remove duplicates, keeping last occurrence and preserving order
-    # e.g. "-lhwloc -lqthread -lhwloc ..." -> "-lqthread -lhwloc ..."
-    seen = set()
-    compile_args_3p_dedup = [arg for arg in reversed(compile_args_3p)
-                             if not (arg in seen or seen.add(arg))]
-    ENV_VALS['  CHPL_THIRD_PARTY_COMPILE_ARGS'] = ' '.join(reversed(compile_args_3p_dedup))
+    ENV_VALS['  CHPL_THIRD_PARTY_COMPILE_ARGS'] = ' '.join(dedup(compile_args_3p))
 
-    seen = set()
-    link_args_3p_dedup = [arg for arg in reversed(link_args_3p)
-                          if not (arg in seen or seen.add(arg))]
-    ENV_VALS['  CHPL_THIRD_PARTY_LINK_ARGS'] = ' '.join(reversed(link_args_3p_dedup))
+    ENV_VALS['  CHPL_THIRD_PARTY_LINK_ARGS'] = ' '.join(dedup(link_args_3p))
+
+    ENV_VALS['  CHPL_THIRD_PARTY_HOST_LINK_ARGS'] = ' '.join(dedup(host_link_args_3p))
 
     ENV_VALS['  CHPL_PE_CHPL_PKGCONFIG_LIBS'] = chpl_llvm.gather_pe_chpl_pkgconfig_libs()
 
+""" Remove duplicates, keeping last occurrence and preserving order
+e.g. "-lhwloc -lqthread -lhwloc ..." -> "-lqthread -lhwloc ..."""
+def dedup(args):
+    seen = set()
+    ret = [arg for arg in reversed(args)
+           if not (arg in seen or seen.add(arg))]
+    return reversed(ret)
```

Comment:
How do we end up with duplicates and did this PR change anything with respect to that, or is that just something you noticed while working on this?

Reply:
Oh, duh -- nevermind. You just moved dedup into a helper function to use it in more places.